### DISCUSSION TITLE
Storage - Convert Video to mp4

### DISCUSSION THREAD

**Discussion by dommarr** [supabase/supabase#3773] [Feature Requests]
Hi,

One idea for storage: automatically convert webm videos to mp4. 

When using WebRTC to record video, many browsers record in webm format. Webm is not playable in Safari, so Webm videos need to be converted to mp4 to playback across all browsers and devices. 

This is a common issue developers face with video and most solutions are expensive. Here are potential solutions:
- [FFmpeg](https://www.ffmpeg.org/) 
- [AWS Elemental MediaConvert](https://aws.amazon.com/mediaconvert/)
- [Mux](https://mux.com/)

**Comment by zakaria-chahboun** [comment]
Easy man! The user just has to use Firefox or Chromium based browsers :laughing:  *(just kidding)*

i guess they already working on something like: 
![image](https://user-images.githubusercontent.com/18173275/140613239-ec320644-3aad-44cf-8a91-481ae6d9f489.png)

**Reply by dommarr** [reply]
Thanks for sharing!

**Reply by sunnysahijwani** [reply]
Is this feature released in Supabase? 
I mean I am trying to include ffmpeg in Edge function like this : - 
import { FfmpegClass } from "https://deno.land/x/deno_ffmpeg@v3.1.0/mod.ts";
but it does not seems to be working. I am sure I am new to it, so I might be missing the important configurations. My simple code looks like this : - 
const ffmpeg = new FfmpegClass();
ffmpeg.addInput("HTTPURLToVideo");
await ffmpeg.videoBitrate("1000k").save("./user-videos/output.mp4");

and I get error saying : -
 "error": "No ffmpeg directory specified!"

and I have no idea what to provide in path if I call setFmfmpegPath function like this at 
ffmpeg.setFfmpegPath("????");


Can anyone please help?

**Comment by novagorge30-cyber** [comment]
Converting videos to MP4 is a practical way to reduce storage usage while improving compatibility across devices and media players. Using the H.264 video codec with AAC audio typically provides an excellent balance between file size and quality.

If you're using Windows and need to convert multiple videos efficiently, SysTools Video Converter Software is another option to consider. It supports batch conversion to MP4 and other popular formats while maintaining video quality, making it useful for managing large video collections.
